### PR TITLE
[XLA:GPU][ROCm] Emit llvm::Intrinsic::nvvm_atomic_load_add_f32 only on NVPTX

### PR DESCRIPTION
**Pull request by whchung**
Emit llvm::Intrinsic::nvvm_atomic_load_add_f32 only on NVPTX

### CODE REVIEW

File: tensorflow/compiler/xla/service/gpu/ir_emitter.cc [hunk 1]
```diff
@@ -205,8 +205,9 @@ bool IrEmitter::MaybeEmitDirectAtomicOperation(
   }
 
   if (root_opcode == HloOpcode::kAdd) {
+    llvm::Triple target_triple = llvm::Triple(module_->getTargetTriple());
     // NVPTX supports atomicAdd on F32 and integer types.
-    if (element_type == F32) {
+    if (target_triple.isNVPTX() && (element_type == F32)) {
```
Comment (thomasjoerg) [inline tensorflow/compiler/xla/service/gpu/ir_emitter.cc]: Nit: No need for extra parenthesis around `element_type == F32`.

Comment (thomasjoerg) [inline tensorflow/compiler/xla/service/gpu/ir_emitter.cc]: Do AMDGPU support atomic intrinsics? Does it make sense to add a TODO to support them here?

**Review by thomasjoerg — DISMISSED** [review]

File: tensorflow/compiler/xla/service/gpu/ir_emitter.cc (hunk 1 above)
Comment (whchung) [reply tensorflow/compiler/xla/service/gpu/ir_emitter.cc]: There are atomic intrinsics but they don't support floating point types. So we really can't put a TODO here.

Comment (whchung) [reply tensorflow/compiler/xla/service/gpu/ir_emitter.cc]: removed.

**Comment by whchung** [comment]
@thomasjoerg I've address your comments in this PR. Could you help re-review again? Thanks. Also I may need your help put `cla: yes` label back on. The bot somehow removed it.

**Review by thomasjoerg — APPROVED** [review]

**Pull request merged**